File: Electronics_Shop_17.Services/Database/CartItem.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class CartItem : IHelperId
    {
        public int Id { get; set; }
        public int Quantity { get; set; }
        public int ProductId { get; set; }
        public virtual Product Product { get; set; }
        public int ShoppingCartId { get; set; }
        public int ProductColorId { get; set; }
        public virtual ProductColor ProductColor { get; set; }
    }
}

File: angular17_web/src/app/models/cart-item/cart.item.ts
import {Product} from '../product/product';
import {ProductColor} from '../product-color/product.color';

export class CartItem {
  id!: number;
  quantity!: number;
  productId!: number;
  product?: Product;
  shoppingCartId!: number;
  finalPrice!: number;
  productColorId!: number;
  productColor?: ProductColor;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.quantity = data.quantity;
    this.productId = data.productId;
    this.product = data.product ? new Product(data.product) : undefined;
    this.shoppingCartId = data.shoppingCartId;
    this.finalPrice = data.finalPrice;
    this.productColorId = data.productColorId;
    this.productColor = data.productColor ? new ProductColor(data.productColor) : undefined;
  }
}
